src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.BlazorWasmCompanionExtension/src/extension.ts

```diff
@@ -0,0 +1,81 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { spawn } from 'child_process';
+import { acquireDotnetInstall } from './acquireDotnetInstall';
+import { getAvailablePort } from "./getAvailablePort";
+
+export function activate(context: vscode.ExtensionContext) {
+    const outputChannel = vscode.window.createOutputChannel("Blazor WASM Debug Proxy");
+    const pidsByUrl = new Map<string, number>();
+
+    const launchDebugProxy = vscode.commands.registerCommand('blazorwasm-companion.launchDebugProxy', async () => {
+        try {
+            const debuggingPort = await getAvailablePort(9222);
+            const debuggingHost = `http://localhost:${debuggingPort}`;
+
+            const debugProxyLocalPath = `${context.extensionPath}/BlazorDebugProxy/BrowserDebugHost.dll`;
+            const spawnedProxyArgs = [debugProxyLocalPath , '--DevToolsUrl', debuggingHost];
+
+            let dotnet = "dotnet";
+            // The vscode.env.remoteName property is set when connected to
+            // any kind of remote workspace. See
+            // https://code.visualstudio.com/api/advanced-topics/remote-extensions#varying-behaviors-when-running-remotely-or-in-the-codespaces-browser-editor
+            const isRemote = vscode.env.remoteName !== 'undefined';
+            if (isRemote) {
```

Comment:
Given that [we'll be trying to boot this server always](https://github.com/dotnet/aspnetcore-tooling/pull/3013#discussion_r567075152) shouldn't we always be trying to acquire `dotnet`?

Asking because if the user is on 3.1 doesn't the expectation that the `dotnet` on the path is "sufficient" fall apart?